### ISSUE TITLE
FR: alternative EnumCache and extensions to De/Humanize enums

### ISSUE THREAD

**Issue by KillyMXI** [Humanizr/Humanizer#1490]
My understanding is that there is currently `DehumanizeTo` extension that is intended to work exactly in reverse of `Humanize`.
At the same time, it is not necessarily what I expect as a user of the library.

```csharp
public enum ExampleEnum
{
    [Display(Name = "The foo", Description = "something about what effect Foo makes")]
    Foo,
    [Display(Name = "The bar", Description = "something about what effect Bar makes")]
    Bar,
}
```

I don't expect to parse input string in the format of description.
I expect the input script to be the best approximation of a parseable name.

And so, I think the reasonable string-to-value dictionary may consist of normalized strings taken from `ToString` value, display name, description, whatever else can be located in the attributes...

Current implementation prioritizes Description:

https://github.com/Humanizr/Humanizer/blob/3d80235a9acf3b32acae65bf2b2d9284c54a20de/src/Humanizer/EnumCache.cs#L64-L73

I don't understand the utility of this. It breaks valid usage scenarios.
And it seems that for a workaround I have to reimplement a custom dehumanizer from scratch.

This is probably related to #700.

By extension, if I need the `Humanize` function - I will face the same issue - I'm not necessarily need the description - I may prefer to:
- use Name attribute if it is available;
- or humanize `ToString` value.

To resolve both problems, alternative implementation to current `EnumCache` and extensions might be conceived, probably in parallel to the existing one to avoid breaking change.
It may be not exactly what I expect it to be to solve each direction issue, but I'd like this to be a started for a discussion at least.

Seems like Humanizer might've been around before `System.ComponentModel.DataAnnotations.DisplayAttribute` and is still based on more limited assumptions.

**Comment by clairernovotny** [comment]
Thank you for the detailed proposal and for calling out the DisplayAttribute semantics. The two actionable contracts are already tracked separately: #700 covers DehumanizeTo aliases, and #926 covers selecting the metadata source used by Humanize. I have carried this report's unique Display.ShortName and normalization acceptance points into those canonical issues, so we can close this broader alternative-cache proposal without losing the useful requirements.
